controllers/osbuildconfig_controller.go

```diff
@@ -47,13 +53,54 @@ type OSBuildConfigReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
 func (r *OSBuildConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling", "edgeDevice", req)
+
+	osBuildConfig, err := r.OSBuildConfigRepository.Read(ctx, req.Name, req.Namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{Requeue: true}, err
+	}
 
-	// TODO(user): your logic here
+	if osBuildConfig.DeletionTimestamp != nil {
+		return ctrl.Result{}, nil
+	}
+
+	err = r.createOSBuildCR(ctx, osBuildConfig, logger)
+	if err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
 
 	return ctrl.Result{}, nil
 }
 
+func (r *OSBuildConfigReconciler) createOSBuildCR(ctx context.Context, osBuildConfig *osbuilderprojectflottaiov1alpha1.OSBuildConfig, logger logr.Logger) error {
+	osBuildCurrentVersion := string(rune(*osBuildConfig.Status.LastVersion + 1))
+
+	osBuildConfigCp := osBuildConfig.DeepCopy()
+	// copy to osbuildCR if not exist
+	osBuild := &osbuilderprojectflottaiov1alpha1.OSBuild{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: osBuildConfigCp.Name + "-" + osBuildCurrentVersion,
+		},
+		Spec: osbuilderprojectflottaiov1alpha1.OSBuildSpec{
+			Details: osBuildConfigCp.Spec.Details,
+			Config:  osbuilderprojectflottaiov1alpha1.OSBuildConfig{},
+		},
+	}
+
+	err := r.OSBuildRepository.Create(ctx, osBuild)
+	if err != nil {
+		logger.Error(err, "cannot create osBuild")
+		return err
+	}
+	logger.Info("A new OSBuild CR was created", osBuild.Name)
+
```

Comment:
I think we need to update the LastVersion field of the OSBuildConfig, no?

Reply:
AFAIN it's happening automatically

Reply:
Sorry, disregard my previous comment. It won't happen automatically. The controller needs to update it. Something like this: https://github.com/ctera/kubefiler-operator/blob/2a06c8bd0f0b9ba04dd783c8d6c4867762fc4f87/internal/resources/kubefiler.go#L186

Reply:
This only updated the local structure, not the object stored in etcd. You need to run `client.Status().Update(ctx, osBuildConfig)`.  See the next line: https://github.com/ctera/kubefiler-operator/blob/2a06c8bd0f0b9ba04dd783c8d6c4867762fc4f87/internal/resources/kubefiler.go#L187